### PR TITLE
gh-89381: Fix invalid signatures of math/cmath.log

### PR DESCRIPTION
**Pull request by skirpichev**
* Issue: gh-89381

### CODE REVIEW

**Comment by hauntsaninja** [comment]
Thanks for the PR! I was curious about why the existing code in mathmodule.c didn't work. It turns out that it is I think easy to fix, which led me to make an alternate PR here: #101115.

I think that PR is a slightly smaller change / closer to what whoever did the original argument clinic for mathmodule.c intended. But maybe c_default is sketchy or support for None is desirable, in which case this PR works too!

**Comment by skirpichev** [comment]
@hauntsaninja, thanks.  I left few notes on the new PR.

**Comment by skirpichev** [comment]
Closed per #101123

File: Doc/library/cmath.rst [hunk 1]
```diff
@@ -89,11 +89,11 @@ Power and logarithmic functions
    logarithms.
 
 
-.. function:: log(x[, base])
+.. function:: log(x, base=None)
 
    Returns the logarithm of *x* to the given *base*. If the *base* is not
-   specified, returns the natural logarithm of *x*. There is one branch cut, from 0
-   along the negative real axis to -∞, continuous from above.
+   specified or is None, returns the natural logarithm of *x*. There is one
```
Comment (mdickinson) [inline Doc/library/cmath.rst:95]: In the interests of readability, I'd suggest reverting the change on this line - I think it's clear from context that "not specified" includes the case of deliberately specifying `None`.

**Comment by skirpichev** [comment]
reopened as #101404

**Pull request closed without merging**